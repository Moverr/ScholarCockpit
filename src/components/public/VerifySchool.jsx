import React, { Component } from 'react';
import PropTypes from 'prop-types';
import InputField from '../../forms/inputs/InputField';
import '../../css/verifyschool.css';
import { NavLink } from 'react-router-dom';
import Alert from '../../helpers/Alert';




import { withRouter } from 'react-router-dom';
//todo: services 
import  {VerifySchoolService} from '../../services/SchoolSeervice'


import store from '../../store/Store';
import { pushHistory } from '../../helpers/Utils';






class VerifySchool extends Component {


      constructor(props) {
            super(props);
            this.state = {
                  schoolname: null,
                  message_status: '-success displaynone',
                  message: ' Wait a minute '
            };


            this.handleChange = this.handleChange.bind(this);
            this.handleSubmit = this.handleSubmit.bind(this);

             





      }

      componentWillMount() {

      }

      componentDidMount() {

      }


      handleSubmit(e) {
            e.preventDefault();


            this.setState({
                  message_status: '-warning  displaynone',
                  message: ' school name is required'
            });

            let school_name = (this.state.schoolname == null) ? null : this.state.schoolname.trim();

            if (school_name == null || school_name.length == 0) {

                  this.setState({
                        message_status: '-warning ',
                        message: ' school name is required'
                  });

                  return;
            }




            //todo: Verify School 
            VerifySchoolService(school_name, (response) => {
                  alert("pass me ")
                  // this.props.history.push("/login");
                  // Move.dispatch(setSchoolName("movers"))

            })

      }


      handleChange(evt) {
            let value = evt.target.value;

            this.setState({
                  ...this.state,
                  [evt.target.name]: value
            });



      }




      render() {
            // const counter = useSelector(state =>state.counterReducer) 


            return (
                  <div className="container-fluid verifyschoolwrapper">
                        <form role="form" onSubmit={this.handleSubmit} >
                              <br />
                              <h3 className="center-align header">Enter Your school name. </h3>
                              <Alert message={this.state.message} className={'alert alert' + this.state.message_status} role={"alert"} />

                              <div className="form-group">

                                    <InputField type="text" className="text form-control verify-textinput" callback={this.handleChange} name="schoolname" value={this.state.schoolName}
                                          placeholder=" " />
                                    <button className="btn btn-primary col-md-12 verify-button gray " type="submit">NEXT</button>

                                    <ul className="nav sublinks">
                                          <li> <NavLink to={'#'} activeClassName="active"> Find School </NavLink>
                                          </li>

                                          <li> <NavLink to={'#'} activeClassName="active"> Register  school   </NavLink>
                                          </li>

                                    </ul>
                              </div>
                        </form>
                  </div>
            );
      }






}

VerifySchool.propTypes = {
};


 
export default withRouter(VerifySchool);
