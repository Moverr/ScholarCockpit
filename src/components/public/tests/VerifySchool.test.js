import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import VerifySchool from '../VerifySchool';
 
 


      
 let wrapper = shallow(<VerifySchool />);

   

describe('Testing Input FIeld', () => {
     

  it('should  render correctly', () => {       

      expect(wrapper).toMatchSnapshot();
  });

   
})



 