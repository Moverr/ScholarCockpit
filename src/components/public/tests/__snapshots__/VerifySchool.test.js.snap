// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Input FIeld should  render correctly 1`] = `
<div
  className="verifySchoolview col-md-12"
>
  <br />
  <h5>
    Enter School Name
  </h5>
  <InputField
    className="text"
    placeholder="Enter School Name "
    type="text"
    value=""
  />
  <br />
  <button>
    CONTINUE
  </button>
</div>
`;
